modules/metalsmith/doc_assets.js

```diff
@@ -0,0 +1,168 @@
+    return new Promise(function (resolve) {
+
+        const file = files[filename];
+
+        const contents = file.contents.toString();
+        const $ = cheerio.load(contents);
+
+        const url_promises = [];
+
+        $(selector).each(function (i, el) {
+
+            const $el = $(el);
+            const attr = get_attribute(el);
+            const url = $el.attr(attr);
+            if (url && url.search(/^nx_asset?\:\/\//) !== -1) {
+                const uid = url.replace(/(^\w+:|^)\/\//, '');
+                const p = nuxeo.repository()
+                    .fetch(uid, {schemas: ['dublincore', 'file', 'document_asset']})
+                    .then(function (doc) {
+
+                        // if there is no file return
+                        if (!doc.properties['file:content']) {
+                            error('nx_asset without file:content');
+                            return null;
+                        }
+
+                        // build the right url
+                        const folder = 'nx_assets';
+                        const ext = /(?:\.([^.]+))?$/.exec(doc.properties['file:content'].name)[1];
+                        const asset_type = doc.properties['doc_asset:nature'];
+                        const asset_file = folder + '/' + uid + '+' + asset_type + '.' + ext;
+                        const href = '/' + asset_file;
+
+                        $el.attr(attr, href);
+                        const html = $.html();
+                        file.contents = new Buffer(html);
+                        files[filename] = file;
+
+                        // if file already exist in fs
+                        if (fs.existsSync(asset_file)) {
+                            const content = fs.readFileSync(asset_file);
+                            const file_digest = checksum(content, doc.properties['file:content'].digestAlgorithm);
+
+                            // if digest matches ignore
+                            if (file_digest === doc.properties['file:content'].digest) {
+                                info('asset: %s already exists, ignoring', uid);
+                                return null;
+                            }
+                        }
+                        return doc.fetchRendition('docAsset')
+                            .then(function (res) {
+                                const writable = fs.createWriteStream(asset_file);
+                                res.body.pipe(writable);
+                                writable.on('finish', function () {
+                                    info('Downloaded asset: %s', uid);
+                                    writable.close();
+                                });
+                            });
+                    })
+                    .catch(function (err) {
+                        throw err;
+                    });
+
+                url_promises.push(p);
+            }
+
+            Promise.all(url_promises)
```

Comment:
No need to create the wrapper promise in the beginning to call `resolve()` here. You can just `return Promise.all([...])`